### PR TITLE
Updates salt-reposync to v6 and corrects salt versions for release yum defs

### PR DESCRIPTION
**Pull request by lorengordon**

### CODE REVIEW

**Comment by lorengordon** [comment]
This doesn't need a new version bump because the last build failed on the master branch (so the tag was never created)

**Review by cahnk — APPROVED** [review]

File: dev/salt-repo/terraform.tfvars [hunk 1]
```diff
@@ -9,6 +10,7 @@ repos = [
   },
   {
     repo_prefix      = "repo/archive/saltstack/salt/linux/"
+    salt_s3_bucket   = "s3"
     salt_s3_endpoint = "https://s3.archive.repo.saltproject.io/"
     yum_prefix       = "yum.defs/saltstack/salt/"
     salt_versions = [
```
Comment (eemperor) [inline dev/salt-repo/terraform.tfvars:16]: Did `3003.3` and `3004` listed here under the archive section from the previous PR actually work?  They're not in Salt's archive location.

File: release/salt-yum-defs/terraform.tfvars [hunk 2]
```diff
@@ -3,17 +3,15 @@ repos = [
...
     ]
   },
   {
     repo_prefix = "repo/archive/saltstack/salt/linux/"
     yum_prefix  = "yum.defs/saltstack/salt/"
     salt_versions = [
-      "2019.2.8",
-      "2019.2.5",
+      "3004",
+      "3003.3",
```
Comment (eemperor) [inline release/salt-yum-defs/terraform.tfvars:14]: This is related to the previous comment.  These versions are not in Salt's archive location from what I can tell.

Comment (lorengordon) [reply release/salt-yum-defs/terraform.tfvars:14]: They are under the `py3` path...

* https://s3.archive.repo.saltproject.io/py3/redhat/7/x86_64/archive/3003.3/
* https://s3.archive.repo.saltproject.io/py3/redhat/7/x86_64/archive/3004/

The `yum` path is only for python2 packages, and these are not published for python2...

Comment (lorengordon) [reply release/salt-yum-defs/terraform.tfvars:14]: salt-reposync pulls _both_ the python2 and python3 packages for each version specified...

https://github.com/plus3it/salt-reposync/blob/08cc9f020395339fe87e023cc74a32e30801d0e3/modules/repo/main.tf#L28


https://github.com/plus3it/salt-reposync/blob/08cc9f020395339fe87e023cc74a32e30801d0e3/modules/repo/main.tf#L36

**Pull request merged**